lib/gce.sh

```diff
@@ -0,0 +1,40 @@
+# TODO generify this function to take a set of IPs
+# Set the internal and external ip arrays for the master node and all minions.
+# Sets: MASTER_IPS, MINION_NAMES, MINION_IPS, and HOSTS global vars.
+# Note: the MINION_IPS map is not yet referenced but remains for future usage.
+function gce_util::master_minions_ips() {
+	echo "-- Setting global master and minion internal and external ip addresses."
+	# format: (internal-ip external-ip)
+	MASTER_IPS=($(gcloud compute instances list --filter="zone:($GCP_ZONE) name:($GK_MASTER_NAME)" \
+		--format="value(networkInterfaces[0].networkIP,\
+		  networkInterfaces[0].accessConfigs[0].natIP)"))
+	if (( ${#MASTER_IPS[@]} == 0 )); then
+		echo "Failed to get master node $GK_MASTER_NAME's ip addresses."
+		return 1
+	fi
+
+	# Make a single call to gcloud to get all minion info
+	# format: (hostname internal-ip external-ip...)
+	local minions=($(gcloud compute instances list --filter="zone:($GCP_ZONE) name:($GK_NODE_NAME)" \
+		--format="value(name,networkInterfaces[0].networkIP,\
+		  networkInterfaces[0].accessConfigs[0].natIP)"))
+	if (( ${#minions[@]} == 0 )); then
+		echo "Failed to get minions."
+		return 1
+	fi
+
+	# set MINION_NAMES and HOSTS arrays, and MINION_IPS map
```

Comment:
may not be needed since `util::get_instance_info` returns a sting which is easily converted to a map. Eg:
`declare -A info=$(util::get_instance_info $filter KEY1 KEY2...)`

Reply:
right, will fix.